File: hosts/jetbrains/dx-intellij-platform/src/main/kotlin/dev/dx/intellij/commands/DxCommandPlans.kt
package dev.dx.intellij.commands

data class DxCommandPlan(
    val actionId: String,
    val operation: String,
    val transport: String,
    val requiresRuntimeProof: Boolean,
    val mutatesProject: Boolean,
)

object DxCommandPlans {
    private val plans = listOf(
        DxCommandPlan(
            actionId = "dev.dx.intellij.showStatus",
            operation = "dx.status",
            transport = "local-service",
            requiresRuntimeProof = true,
            mutatesProject = false,
        ),
        DxCommandPlan(
            actionId = "dev.dx.intellij.searchAssets",
            operation = "dx.assets.search",
            transport = "local-service",
            requiresRuntimeProof = true,
            mutatesProject = false,
        ),
        DxCommandPlan(
            actionId = "dev.dx.intellij.showReceipts",
            operation = "receipt.showPath",
            transport = "host-ui",
            requiresRuntimeProof = false,
            mutatesProject = false,
        ),
    ).associateBy { it.actionId }

    fun forAction(actionId: String): DxCommandPlan? = plans[actionId]
}

File: hosts/visual-studio/dx-visual-studio/src/CommandPlans/DxCommandPlans.cs
using System.Collections.Generic;
using Dx.VisualStudio.CommandCenter.Commands;

namespace Dx.VisualStudio.CommandCenter.CommandPlans;

internal static class DxCommandPlans
{
    private static readonly IReadOnlyDictionary<int, DxCommandPlan> Plans =
        new Dictionary<int, DxCommandPlan>
        {
            [CommandIds.ShowStatus] = new(
                CommandIds.ShowStatus,
                Operation: "dx.status",
                Transport: "local-service",
                RequiresRuntimeProof: true,
                MutatesSolution: false),
            [CommandIds.SearchAssets] = new(
                CommandIds.SearchAssets,
                Operation: "dx.assets.search",
                Transport: "local-service",
                RequiresRuntimeProof: true,
                MutatesSolution: false),
            [CommandIds.ShowReceipts] = new(
                CommandIds.ShowReceipts,
                Operation: "receipt.showPath",
                Transport: "host-ui",
                RequiresRuntimeProof: false,
                MutatesSolution: false)
        };

    internal static DxCommandPlan ForCommand(int commandId) => Plans[commandId];
}
